presto-hive/src/main/java/io/prestosql/plugin/hive/util/SerDeUtils.java

```diff
@@ -312,11 +321,22 @@ private static long formatDateAsLong(Object object, DateObjectInspector inspecto
         return inspector.getPrimitiveJavaObject(object).toEpochDay();
     }
 
-    private static long formatTimestampAsLong(Object object, TimestampObjectInspector inspector)
+    private static DecodedTimestamp formatTimestamp(TimestampType type, Object object, TimestampObjectInspector inspector)
     {
+        long seconds;
+        int nanos;
+
         if (object instanceof TimestampWritable) {
```

Comment:
This check was already in the class, but when does it come up? `SerDeUtils` is only used to read structural types, in `GenericHiveRecordCursor`, and that class always assumes that the object can be read by the associated inspector, as in the `else` block a few lines down.